CryptoBinPkg/Plugin/ReportCrypto/ReportCrypto.py

```diff
@@ -0,0 +1,202 @@
+## @file BundleCrypto.py
+# Plugin to Bundle the CryptoBin build output
+#
+##
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+##
+from edk2toolext.environment.plugintypes.uefi_build_plugin import IUefiBuildPlugin
+from pathlib import Path
+from datetime import datetime
+from git import Repo
+import lzma
+
+# Dictionaries
+binary_to_type = {}
+arch_and_type_to_lib = {}
+
+# defines
+offset = 40
+
+class ReportCrypto(IUefiBuildPlugin):
+    """
+    ReportCrypto is a plugin class that generates a report about the crypto binaries built during the build process.
+    The report includes the next information:
+    - Submodules information
+    - Built binaries sizes
+    - Linked OpenSSL library for each binary
+    - OpenSSL library flags
+    Methods:
+    --------
+    do_post_build(thebuilder):

```

Comment:
function documentation must be assocaited with the method itself, not in the overall class documentation. Here is a random example: https://github.com/tianocore/edk2-pytool-extensions/blob/26f14583fbfbbe9736fe941147ed1a791b8b7ffd/edk2toolext/environment/plugin_manager.py#L41